applications/RomApplication/custom_strategies/rom_builder_and_solver.h

```diff
@@ -676,6 +410,347 @@ class ROMBuilderAndSolver : public BuilderAndSolver<TSparseSpace, TDenseSpace, T
+            } else {
+                p_condition->SetValue(HROM_WEIGHT, 1.0);
+            }
+        });
+
+        // Dequeueing elements
+        std::size_t err_id;
+        mSelectedElements.reserve(element_queue.size_approx());
+        Element::Pointer p_element;
+        while ( (err_id = element_queue.try_dequeue(p_element)) != 0) {
+            mSelectedElements.push_back(std::move(p_element));
+        }
+
+        // Dequeueing conditions
+        mSelectedConditions.reserve(condition_queue.size_approx());
+        Condition::Pointer p_condition;
+        while ( (err_id = condition_queue.try_dequeue(p_condition)) != 0) {
+            mSelectedConditions.push_back(std::move(p_condition));
+        }
+
+        // Wrap-up
+        mHromSimulation = !(mSelectedElements.empty() && mSelectedConditions.empty());
+        mHromWeightsInitialized = true;
+
+        rModelPart.GetCommunicator().GetDataCommunicator().OrReduceAll(mHromSimulation);
+
+        KRATOS_CATCH("")
+    }
+    
+    static DofQueue ExtractDofSet(
+        typename TSchemeType::Pointer pScheme,
+        ModelPart& rModelPart)
+    {
+        KRATOS_TRY
+
+        DofQueue dof_queue;
+
+        // Emulates ConcurrentQueue::enqueue_bulk adding move semantics to avoid atomic ops
+        const auto enqueue_bulk_move = [](DofQueue& queue, DofsVectorType& dof_list) {
+            for(auto& p_dof: dof_list) {
+                queue.enqueue(std::move(p_dof));
+            }
+            dof_list.clear();
+        };
+
+        // Inspecting elements
+        DofsVectorType tmp_dof_list; // Preallocation
+        block_for_each(rModelPart.Elements(), tmp_dof_list,
+            [&](const Element& r_element, DofsVectorType& dof_list)
+        {
+            pScheme->GetDofList(r_element, dof_list, rModelPart.GetProcessInfo());
+            enqueue_bulk_move(dof_queue, dof_list);
+        });
+
+        // Inspecting conditions
+        block_for_each(rModelPart.Conditions(), tmp_dof_list,
+            [&](const Condition& r_condition, DofsVectorType& dof_list)
+        {
+            pScheme->GetDofList(r_condition, dof_list, rModelPart.GetProcessInfo());
+            enqueue_bulk_move(dof_queue, dof_list);
+        });
+
+        // Inspecting master-slave constraints
+        std::pair<DofsVectorType, DofsVectorType> tmp_ms_dof_lists; // Preallocation
+        block_for_each(rModelPart.MasterSlaveConstraints(), tmp_ms_dof_lists,
+            [&](const MasterSlaveConstraint& r_constraint, std::pair<DofsVectorType, DofsVectorType>& dof_lists)
+        {
+            r_constraint.GetDofList(dof_lists.first, dof_lists.second, rModelPart.GetProcessInfo());
+
+            enqueue_bulk_move(dof_queue, dof_lists.first);
+            enqueue_bulk_move(dof_queue, dof_lists.second);
+        });
+
+        return dof_queue;
+
+        KRATOS_CATCH("")
+    }
+
+    static DofsArrayType SortAndRemoveDuplicateDofs(DofQueue& dof_queue)

```

Comment:
```suggestion
    static DofsArrayType SortAndRemoveDuplicateDofs(DofQueue& rDofQueue)
```
Please follow style guide.